### PR TITLE
Restore defaultForCurrency variable in the Card

### PR DESCRIPTION
**Pull request by Maxim-Inv**

### CODE REVIEW

**Comment by Andrewangeta** [comment]
@Maxim-Inv Where is this property on the card resource? I'm not seeing it here https://stripe.com/docs/api/cards/object

**Comment by Maxim-Inv** [comment]
@Andrewangeta  Oh, really. I still get this field because I haven't upgraded the API version. I'll try to update it now.

**Comment by Maxim-Inv** [comment]
@Andrewangeta I've checked Stripe API version 2023-10-16, the `default_for_currency` field is present in the response.
```
  {
      "id": "card_......",
      "object": "card",
      "account": "acct_......",
      "address_city": null,
      "address_country": null,
      "address_line1": null,
      "address_line1_check": null,
      "address_line2": null,
      "address_state": null,
      "address_zip": null,
      "address_zip_check": null,
      "available_payout_methods": [
        "standard",
        "instant"
      ],
      "brand": "MasterCard",
      "country": "US",
      "currency": "usd",
      "cvc_check": "pass",
      "default_for_currency": true,
      "dynamic_last4": null,
      "exp_month": 12,
      "exp_year": 2025,
      "fingerprint": "......",
      "funding": "debit",
      "last4": "0000",
      "metadata": {},
      "name": null,
      "status": "new",
      "tokenization_method": null,
      "wallet": null
    },
```

**Comment by Maxim-Inv** [comment]
@Andrewangeta actually, this variable was in the previous versions and I use it in my project. So, it would be great if you accept this PR.

File: Sources/StripeKit/Billing/Invoices/Invoice.swift [hunk 1]
```diff
@@ -318,6 +320,11 @@ public struct Invoice: Codable {
     }
 }
 
+public struct SubscriptionDetails: Codable {
+	/// Set of key-value pairs that you can attach to an object. This can be useful for storing additional information about the object in a structured format.
+	public var metadata: [String: String]?
```
Comment (Andrewangeta) [inline Sources/StripeKit/Billing/Invoices/Invoice.swift:325]: @Maxim-Inv Can you add a public initializer for this type?

File: Sources/StripeKit/Payment Methods/Cards/Card.swift [hunk 2]
```diff
@@ -32,6 +32,8 @@ public struct Card: Codable {
     @Expandable<Customer> public var customer: String?
     /// If a CVC was provided, results of the check: `pass`, `fail`, `unavailable`, or `unchecked`.
     public var cvcCheck: CardValidationCheck?
+	/// Whether this card is the default external account for its currency.
+	public var defaultForCurrency: Bool?
```
Comment (Andrewangeta) [inline Sources/StripeKit/Payment Methods/Cards/Card.swift:36]: @Maxim-Inv  Can you add this parameter to the initializer?

File: Sources/StripeKit/Billing/Invoices/Invoice.swift [hunk 3]
```diff
@@ -40,6 +40,8 @@ public struct Invoice: Codable {
     public var status: InvoiceStatus?
     /// The subscription that this invoice was prepared for, if any.
     @Expandable<Subscription> public var subscription: String?
+	/// Details about the subscription that created this invoice.
+	public var subscriptionDetails: SubscriptionDetails?
```
Comment (Andrewangeta) [inline Sources/StripeKit/Billing/Invoices/Invoice.swift:44]: @Maxim-Inv Can you add this param to the initializer?

**Review by Andrewangeta — CHANGES_REQUESTED** [review]

File: Sources/StripeKit/Payment Methods/Cards/Card.swift (hunk 2 above)
Comment (Maxim-Inv) [reply Sources/StripeKit/Payment Methods/Cards/Card.swift:36]: Yes

File: Sources/StripeKit/Billing/Invoices/Invoice.swift (hunk 1 above)
Comment (Maxim-Inv) [reply Sources/StripeKit/Billing/Invoices/Invoice.swift:327]: Yes

File: Sources/StripeKit/Billing/Invoices/Invoice.swift [hunk 4]
```diff
@@ -318,6 +322,15 @@ public struct Invoice: Codable {
     }
 }
 
+public struct SubscriptionDetails: Codable {
+	/// Set of key-value pairs that you can attach to an object. This can be useful for storing additional information about the object in a structured format.
+	public var metadata: [String: String]?
+	
+	public init(metadata: [String: String]?) {
```
Comment (Andrewangeta) [inline Sources/StripeKit/Billing/Invoices/Invoice.swift:329]: @Maxim-Inv One last thing, add a default nil param for this and it should be good to merge and tag anew release

Comment (Maxim-Inv) [reply Sources/StripeKit/Billing/Invoices/Invoice.swift]: Ok. Ready.

**Review by Andrewangeta — APPROVED** [review]

**Comment by Maxim-Inv** [comment]
Thank you!

**Pull request merged**